code/obj/item/grenades.dm

```diff
@@ -934,7 +935,10 @@ ADMIN_INTERACT_PROCS(/obj/item/gimmickbomb, proc/arm, proc/detonate)
 	proc/beep(i)
 		var/k = i/2
 		sleep(k*k)
-		flick(icon_state+"_beep", src)
+		if (src.icon_state_armed)
+			src.icon_state = icon_state_armed
```

Comment:
The goal here should be to alternate iconstates based on the original (broken) code

 ```suggestion
		if (icon_state_armed)
			if (icon_state == src::icon_state)
				icon_state = icon_state_armed
			else
				icon_state = src::icon_state
```

Reply:
Got it, though I had to replace the ::'s with .'s because if not the armed animation wouldn't stay looped properly otherwise. I assume that was just a (typo?) in the suggestion code.

Reply:
`::` is the scope operator used here to retrieve initial values. The problem here is that some of the bombs rely on alternating between their unarmed and armed icons, while as others expect to be left in armed icon state. The best fix is to animate the armed icons

Reply:
> `::` is the scope operator used here to retrieve initial values. The problem here is that some of the bombs rely on alternating between their unarmed and armed icons, while as others expect to be left in armed icon state. The best fix is to animate the armed icons

So should I change it to add the scope operator :: back in like in your suggestion code, or is the code fine the way I have it right now? I don't see in the code where some bombs "rely on alternating" and some "expect to stay in armed state" so I'm not really sure what you are referring to? The beep code only affects Gimmick Bombs, of which there are only four main bombs under that subtype, and I don't see anything making them act any different when undergoing the Beep proc.

Reply:
when armed many items such as grenades have an animated icon that flashes to make it obvious. The gimmick bombs dont because it was long since bugged. The original flick didnt do anything. If you just want to have them sit on the armed icon state with a static icon then theres no need for the code to flip the icon back and forth each beep, Though the code now has pointless checks like icon_state == icon_state which would never fail